packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditGenerators.ts

```diff
@@ -21,45 +21,65 @@ import {
 	rootFieldKeySymbol,
 	UpPath,
 } from "../../../core";
+import { Multiplicity } from "../../../feature-libraries";
 
-export type Operation = TreeEdit | Synchronize;
+export type Operation = TreeEdit | Synchronize | TransactionEdit;
 
 export interface TreeEdit {
 	type: "edit";
-	contents: FuzzChange;
-	/** index of the tree to apply the edit to. */
-	index: number;
+	contents: FuzzEdit;
+}
+
+export interface Synchronize {
+	type: "synchronize";
+}
+
+export interface TransactionEdit {
+	type: "transaction";
+	contents: FuzzTransactionEdit;
+	treeIndex: number;
+}
+
+export type FuzzEdit = FuzzFieldEdit | FuzzNodeEdit;
+
+export type FuzzFieldChange = FuzzInsert | FuzzDelete;
+
+export interface FuzzFieldEdit {
+	editType: "fieldEdit";
+	change: FuzzFieldChange;
 }
 
 export interface FuzzInsert {
-	fuzzType: "insert";
+	fieldEditType: "insert";
+	fieldKind: Multiplicity;
```

Comment:
I just saw this (and the one on `FuzzDelete`) which explains where the part about specifying the field kind went.

When I wrote "we want a separate flag to describe which kind of field it is an edit of, then for each field kind (currently just sequence) we want a flag to describe whether it is an insert or a delete." in the previous round of review, I meant for those to be different levels of the hierarchy (as is the case for the other flags). Sorry that was unclear.

So `FuzzInsert` and `FuzzDelete` shoudn't get a `fieldKind: Multiplicity` flag. If you look at the APIs for optional and value field you'll see that they don't have "insert" and "delete" methods, so having a `FuzzInsert` with a `Mutiplicity.optional` on it doesn't really represent how our APIs are structured. Also there's the problem that we can have many field kind implementations for a single multiplicity, which this approach would struggle to model.

Hopefully my other comment (the one that starts with "I'm not seeing the layer at which...") clarifies what I had in mind, but if you have any doubt, let me know.